### ISSUE TITLE
Fails to build for watchOS using Xcode 10 from Carthage

### ISSUE THREAD

**Issue by NachoSoto** [ReactiveCocoa/ReactiveCocoa#3602]
**Cartfile**:
```
github "ReactiveCocoa/ReactiveCocoa" "xcode-10"
```

**Command:**
`$ carthage update ReactiveCocoa --platform watchos`

-------

This results in this command:
```
/usr/bin/xcrun xcodebuild -workspace /project/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa.xcworkspace -scheme ReactiveMapKit -configuration Release -derivedDataPath /Users/nachosoto/Library/Caches/org.carthage.CarthageKit/DerivedData/10.0_10L176w/ReactiveCocoa/36f0383edcc2de9431baaf4f75a1f8be201176cb -sdk watchsimulator -destination platform=watchOS\ Simulator,id=F192C9E9-FAA3-4FBD-B755-8BE25DCA115B -destination-timeout 3 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES build
```

And that fails with a bunch of errors like this because of missing `UIKit` classes (which shouldn't be compiled on the watchOS framework):
```
/project/Carthage/Checkouts/ReactiveCocoa/ReactiveCocoa/UIKit/ReusableComponents.swift:15:11: error: use of undeclared type 'UITableViewCell'
extension UITableViewCell: Reusable {}
          ^~~~~~~~~~~~~~~
```

Not sure what's going on honestly, because `ReactiveMapKit` builds correctly from Xcode.

**Comment by mdiep** [comment]
We should probably switch to the `canImport` checks when using Swift 4.2. That might clear these up?

**Comment by ikesyo** [comment]
I assume this is due to the fact that `ReactiveMapKit` is a universal-framework target. Since MapKit is not available on watchOS, removing `watchos` and `watchsimulator` from `ReactiveMapKit`'s `SUPPORTED_PLATFORMS` build setting would fix the issue.

**Comment by ikesyo** [comment]
Should be addressed by #3604.
